keras/src/layers/preprocessing/image_preprocessing/equalization.py

```diff
@@ -0,0 +1,193 @@
+    (independently of which backend you're using).
+
+    Args:
+        value_range: Optional list/tuple of 2 floats specifying the lower
+            and upper limits of the input data values. Defaults to `[0, 255]`.
+            If the input image has been scaled, use the appropriate range
+            (e.g., `[0.0, 1.0]`). The equalization will be scaled to this
+            range, and output values will be clipped accordingly.
+        bins: Integer specifying the number of histogram bins to use for
+            equalization. Defaults to 256, which is suitable for 8-bit images.
+            Larger values can provide more granular intensity redistribution.
+
+    Inputs: 3D (HWC) or 4D (NHWC) tensor, with float or int dtype. Input pixel
+        values can be of any range (e.g. `[0., 1.)` or `[0, 255]`)
+
+    Output: 3D (HWC) or 4D (NHWC) tensor with intensities redistributed to
+        enhance contrast. The output will be of the same shape and dtype as
+        the input, with values scaled and clipped to the specified
+        `value_range`.
+
+    Example:
+
+    ```python
+    # Create an equalization layer for standard 8-bit images
+    equalizer = keras.layers.Equalization()
+
+    # An image with uneven intensity distribution
+    image = [...] # your input image
+
+    # Apply histogram equalization
+    equalized_image = equalizer(image)
+
+    # For images with custom value range
+    custom_equalizer = keras.layers.Equalization(
+        value_range=[0.0, 1.0],  # for normalized images
+        bins=128  # fewer bins for more subtle equalization
+    )
+    custom_equalized = custom_equalizer(normalized_image)
+    ```
+    """
+
+    def __init__(self, value_range=(0, 255), bins=256, **kwargs):
+        super().__init__(**kwargs)
+        self.bins = bins
+        self._set_value_range(value_range)
+
+    def _set_value_range(self, value_range):
+        if not isinstance(value_range, (tuple, list)):
+            raise ValueError(
+                self._VALUE_RANGE_VALIDATION_ERROR
+                + f"Received: value_range={value_range}"
+            )
+        if len(value_range) != 2:
+            raise ValueError(
+                self._VALUE_RANGE_VALIDATION_ERROR
+                + f"Received: value_range={value_range}"
+            )
+        self.value_range = sorted(value_range)
+
+    def _custom_histogram_fixed_width(self, values, value_range, nbins):
+        values = self.backend.cast(values, "float32")
+        value_min, value_max = value_range
+        value_min = self.backend.cast(value_min, "float32")
+        value_max = self.backend.cast(value_max, "float32")
+
+        scaled = (values - value_min) * (nbins - 1) / (value_max - value_min)
+        indices = self.backend.cast(scaled, "int32")
+        indices = self.backend.numpy.clip(indices, 0, nbins - 1)
+
+        if backend.backend() == "jax":
+            # for JAX bincount is never jittable because of output shape
```

Comment:
Can you provide more details? Having an op that isn't jittable seems like a bug in the top. Also, this should be `self.backend.backend()` rather than `backend.backend()`

Reply:
`self.backend.backend()` this fails giving attribute error : `AttributeError: module 'keras.src.backend.tensorflow' has no attribute 'backend'`

Reply:
For JAX's `bincount`, the output shape is dynamic and determined by the highest value in the input array `x`.

its also commented here : https://github.com/keras-team/keras/blob/dbf9878e2c64b92e15b8a221cf999d7b7901c03d/keras/src/backend/jax/numpy.py#L26